src/labels/CustomLabels.labels

```diff
@@ -2015,6 +2015,86 @@
         <shortDescription>Recurring Donation Entry Form message when record is inserted</shortDescription>
         <value>Recurring Donation &quot;{0}&quot; was created.</value>
     </labels>
+    <labels>
+        <fullName>RD2_EntryFormMissingPermissions</fullName>
+        <categories>Recurring-Donations, Entry-Form</categories>
+        <language>en_US</language>
+        <protected>true</protected>
+        <shortDescription>Recurring Donation Entry Form Field Permissions Error</shortDescription>
+        <value>You are missing read and/or edit access to one or more required Recurring Donation fields. Please check with your system administrator.</value>
```

Comment:
@sam-knox Sorry. One more new label to review. This message is displayed if the User is missing Read permissions to any of the required field in the custom UI - Account, Contact, Amount, Period, Frequency, etc. There are far too many to figure out which one(s) are missing and dynamically include in the error, so I kept it generic.

Reply:
@force2b - That's ok, I don't think we need to list all the fields in the error message. Here's a slightly revised version:

"You don&apos;t have the required Recurring Donations field permissions. Contact your system administrator for assistance."

We may need to document what the required fields are though. And the minimum is Read permission - so if a user only has that permission, I assume they would be able to load the custom UI but not make any edits?

Reply:
@sam-knox Thanks. Label change made. That error is only displayed when the User has no permissions to fields that are built into the UI (see list below). There is no error if they have read permissions to the field, but not edit. In that case, the field simply appears in the UI as not editable.

Fields requiring Read access:
- Account
- Contact
- Recurring Type
- Amount
- Installment Period
- Installment Frequency
- Day of Month
- Effective Date
- Payment Method
- Campaign

Reply:
Typing up that list made me think about these fields more. It's impossible that some customers may want to exclude Campaign or Payment Method as field by removing access to them to end users. So, maybe there's alternative route to handle permissions. Rather than throw an exception/error if any of these fields is not accessible, I could have the page simply ignore the error and not render the fields that are not accessible. Of course, a user without read access to many of these fields will be unable to create a new RD and possibly Edit an RD, but then they'll just get a validation error for a missing field and those are already pretty well defined. For example, if they do not have Read access to DayOfMonth, they'd get an error trying to create an RD with an Installment Period of Monthly stating that a DayOfMonth value is required for a Monthly RD.

What do you think?

@npsp-reedestockton @andrewyu-salesforce

Reply:
I think it makes sense to not show the field if the user does not have Edit access to the field. If the user does not have the proper permission to create an RD, then it's up to the Admin to give them the right access. With this if the field is pre-populate and the user wants to update the record, the entry form will still work.